src/Core/Tokens/Tokenable.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Text.Json;
+
+namespace Bit.Core.Tokens
+{
+    public abstract class Tokenable : ITokenable
```

Comment:
I wonder if we really need the interface `ITokenable` or could just go use `Tokenable`? We probably never want to allow any `ITokenable` interface to define their own `ToToken` since we statically call `FromToken`.

Another alternative would be to move the serialize and deserialize logic to the factories but not quite happy with that either.

Reply:
I would like `Token` and `Tokenable` to take care of conversion. I think you're right that the `Tokenable` interface isn't _needed_ anywhere and actually breaks for deserialization -- good catch!